### PR TITLE
Log ota message to MQTT

### PR DESCRIPTION
**Pull request by tim-devel**
I have tweaked code to log some messages to mqtt log

### CODE REVIEW

**Comment by timstanley1985** [comment]
#2921

File: lib/extension/otaUpdate.js [hunk 1]
```diff
@@ -37,7 +37,9 @@ class OTAUpdate extends BaseExtension {
         assert(device != null && device.type === 'device', 'Device not found or not a device');
 
         if (!device.mapped || !device.mapped.ota) {
-            logger.error(`Device '${device.name}' does not support OTA updates`);
+            const message=`Device '${device.name}' does not support OTA updates`;
+            logger.error(message);
+            this.mqtt.log('OTA_update', message);
```
Comment (carldebilly) [inline lib/extension/otaUpdate.js]: You should log something which can be interpreted by external tools, like:
``` json
{"status":"failed", "message":"Device MYDEVICE does not support OTA updates."}
```
Where `status` would be something like `success`, `in_progress` or `failed`.

Comment (Koenkk) [reply lib/extension/otaUpdate.js]: This can go into the `meta` (e.g. `this.mqtt.log('ota_update', message, {status: "in_progress"});`

Comment (carldebilly) [reply lib/extension/otaUpdate.js]: Should also contain the device name in the `meta`... to be able to follow concurrent upgrades (it's supported by Z2M?)

Comment (Koenkk) [reply lib/extension/otaUpdate.js]: Yes that also should be added

Comment (carldebilly) [reply lib/extension/otaUpdate.js]:
```suggestion
            this.mqtt.log('OTA_update', message, {status: "failed", device: device.name);
```

Comment (timstanley1985) [reply lib/extension/otaUpdate.js]: Thanks, I have amended MQTT log messages

**Review by carldebilly — APPROVED** [review]

**Comment by timstanley1985** [comment]
@Koenkk any comments or is this ready for merge?

**Comment by timstanley1985** [comment]
Oops, closed on accident.  Ready for merge?

**Comment by Koenkk** [comment]
yes, thanks!

**Pull request merged**